java-frontend/src/main/java/org/sonar/java/se/checks/StreamConsumedCheck.java

```diff
@@ -75,7 +76,22 @@ public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
     if (syntaxNode.is(Tree.Kind.METHOD_INVOCATION)) {
       return handleMethodInvocation(context, syntaxNode);
     }
-    return context.getState();
+    if (syntaxNode.is(Tree.Kind.NEW_CLASS)) {
+      return removeConstraintOnArgs(context.getState(), syntaxNode);
+    }
+    ProgramState state = context.getState();
+    if (state.peekValue() instanceof SymbolicValue.ExceptionalSymbolicValue) {
+      state = removeConstraintOnExceptionalPath(context);
+    }
+    return state;
+  }
+
+  private static ProgramState removeConstraintOnExceptionalPath(CheckerContext context) {
```

Comment:
matter of naming : this method does not need to know it is working on Exceptional Path. I would go for : `ProgramState removeNotConsumedConstraint(ProgramState state)`